core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt

```diff
@@ -0,0 +1,28 @@
+package fr.sncf.osrd.api.api_v2.conflicts
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import fr.sncf.osrd.api.ConflictDetectionEndpoint
+import fr.sncf.osrd.api.api_v2.DateAdapter
+import java.time.ZonedDateTime
+
+class ConflictDetectionResponse(
+    val conflicts: Collection<Conflict>,
+)
+
+class Conflict(
+    @Json(name = "train_ids") val trainIds: Collection<Long>,
+    @Json(name = "start_time") val startTime: ZonedDateTime,
+    @Json(name = "end_time") val endTime: ZonedDateTime,
+    @Json(name = "conflict_type")
+    val conflictType: ConflictDetectionEndpoint.ConflictDetectionResult.Conflict.ConflictType,
+)
+
+val conflictResponseAdapter: JsonAdapter<ConflictDetectionResponse> =
+    Moshi.Builder()
+        .addLast(DateAdapter())
+        .addLast(KotlinJsonAdapterFactory())
+        .build()
+        .adapter(ConflictDetectionResponse::class.java)
```

Comment:
I haven't tested, but I don't think the distances will be handled properly (missing unit adapter)

Reply:
There are no distances/time deltas here, so the unit adapter is not necessary. However, it will become so when i pute the date adapter in it :)